File: C# Advanced/Strings-And-Text-Processing-Homework/CorrectBrackets/CorrectBrackets.cs
// Write a program to check if in a given expression the ( and ) brackets are put correctly.

using System;

class CorrectBrackets
{
    private static bool AreBracketsCorrect(string expression)
    {
        int openBrackets = 0;
        int closeBrackets = 0;
        for (int i = 0; i < expression.Length; i++)
        {
            if (expression[i] == '(')
            {
                openBrackets++;
            }
            else if (expression[i] == ')')
            {
                if (closeBrackets + 1 > openBrackets)
                {
                    return false;
                }

                closeBrackets++;
            }
        }

        bool areCorrect = openBrackets == closeBrackets;
        return areCorrect;
    }

    static void Main()
    {
        string expression = Console.ReadLine();

        bool areBracketsCorrect = AreBracketsCorrect(expression);
        if (areBracketsCorrect)
        {
            Console.WriteLine("Correct");
        }
        else
        {
            Console.WriteLine("Incorrect");
        }
    }
}

File: JavaScript Basics/Strings/correct-brackets.js
function solve(params) {
    let input = params[0],
        openBrackets = 0,
        closeBrackets = 0;

    for (let i = 0; i < input.length; i += 1) {
        let char = input[i];
        if (char === `(`) {
            openBrackets += 1;
        } else if (char === `)`) {
            if (openBrackets === 0) {
                return `Incorrect`;
            }

            closeBrackets += 1;
        }
    }

    return openBrackets === closeBrackets ? `Correct` : `Incorrect`;
}

console.log(solve([')(a+b))']));